### DISCUSSION TITLE
Adding reservation emails in csv mass upload

### DISCUSSION THREAD

**Discussion by orgalima** [liedekef/events-made-easy#58] [Q&A] [answered]
Hi Franky,
I am uploading events via csv mass upload, which works very well and saves a lot of time. To make this a little more automated, I would like to import reservation emails as well, e.g. pending booking, in the csv. This works with the booking form already (event_registration_form_format), but I did not find the respective variable names for booking emails. 

Is it possible to upload reservation mails via csv mass upload and if so, what are their variable names?

Thanks a lot!
Helge

**Comment by liedekef** [comment]
If I understand your question, you're trying to import bookings, correct? Importing bookings is not possible, due to the event id, answers to custom fields etc ...

**Reply by orgalima** [reply]
Sorry for being imprecise. No, I am trying to import events with a csv that contains columns such as event_name, event_start_date etc. This generally works fine for me. What I would like to add some columns to the csv, so that I dont have to do it manually: I want to import generic reservation emails, like the email that will automatically be sent as a confirmation for a reservation. More precisely I would like to add email bodies and subject emails for reservation confirmation,  payment confirmation, Waiting List notice and reservation cancelled.

**Comment by liedekef** [comment, marked as answer]
Well, that is already possible. From the doc on CSV imports at https://www.e-dynamics.be/wordpress/category/documentation/19-csv-import/csv-import-events-and-locations/:

 _The first line should indicate the names of the columns being imported. Currently the names that are recognized (there are just too many columns to name them all) **can best be found in the function eme_new_event in eme_events.php**. Some (but not all) are: “event_name”,”event_status”,”event_start_date”,”event_start_time”,”event_end_date”,”event_end_time”,”event_notes” and “event_single_event_format”._

And further down:

_Events and locations also have a lot of properties (in fact: those are just a new way of adding event properties without adding extra db columns). These **can be found in the functions eme_init_event_props** (in eme_events.php) and eme_init_location_props (in eme_locations.php).
Those can be imported by prepending the relevant property with ‘prop_’ (e.g. ‘prop_min_allowed’)._

So, it is possible to import just about anything for an event, but you'll have trouble with the carriage returns in your email bodies (csv is not really meant for that).

**Reply by orgalima** [reply]
Thank you, I now understood to work with props, this is a huge upgrade to safe time. Is it possible to also include dynamic fields (see screenshot) in the mass upload of events, without having to add it manually to each event? 

![Screenshot from 2023-02-06 18-13-17](https://user-images.githubusercontent.com/79452113/217038786-e4b1d5d5-6cca-4f6a-ad0b-127c73b84f9d.png)

**Reply by liedekef** [reply]
That info is in the property rsvp_dyndata , but it is json-encoded info and very difficult to import based on CSV info, so: check a real event but I don't recommend doing this.

**Comment by orgalima** [comment]
OK got that, we will do it manually. One more question: I tried hard but couldn't identify what's the name of the template for the email that confirms the reservation (i.e. the first email a person gets when they reserve a space, which in the menu is the first mail, see screenshot)

![Screenshot from 2023-02-07 12-18-44](https://user-images.githubusercontent.com/79452113/217231015-15cc7e9a-6622-4652-a32c-602e04ecea3f.png)

I searched in eme_events.php and tried several variables; also looked in the link you provided without success.

**Comment by liedekef** [comment]
Check the event functions eme_new_event and eme_init_event_props. For older code, the body is the db column event_respondent_email_body , but for a template id it is the property 'event_respondent_email_body_tpl'. Do an inspect on the html and you'll always find the corresponding names.

**Reply by orgalima** [reply]
Thank you Franky, now it Works perfectly!
